src/login/oidc/ClientRegistrar.ts

```diff
@@ -22,70 +22,135 @@
 import { inject, injectable } from "tsyringe";
 import { IFetcher } from "../../util/Fetcher";
 import IClient from "./IClient";
-import ILoginOptions from "../ILoginOptions";
 import IIssuerConfig from "./IIssuerConfig";
+import { IStorageUtility } from "../../storage/StorageUtility";
+
+export interface IRegistrarOptions {
+  sessionId: string;
+  clientId?: string;
+  clientSecret?: string;
+  clientName?: string;
+  redirectUrl?: string;
+}
 
 export interface IClientRegistrar {
   getClient(
-    options: ILoginOptions,
+    options: IRegistrarOptions,
     issuerConfig: IIssuerConfig
   ): Promise<IClient>;
 }
 
 @injectable()
 export default class ClientRegistrar implements IClientRegistrar {
-  constructor(@inject("fetcher") private fetcher: IFetcher) {}
+  constructor(
+    @inject("fetcher") private fetcher: IFetcher,
+    @inject("storageUtility") private storageUtility: IStorageUtility
+  ) {}
 
-  private async performDynamicRegistration(
-    options: ILoginOptions,
+  async getClient(
```

Comment:
Seems like a really big method - I'd consider splitting it up to aid readability (and to ease testing the 'pieces' of it, I'd assume!).

Reply:
Good point. I'll create a ticket for that (I'd like to get the current PR done to be able to move on more reasonably scoped changes)